File: app/src/main/java/com/tripplanner/di/builder/ActivityBuilder.java
package com.tripplanner.di.builder;

import com.tripplanner.ui.trip.TripActivity;
import com.tripplanner.ui.trip.TripActivityModule;
import com.tripplanner.ui.trip.details.TripDetailsProvider;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by suyashg on 09/06/18.
 */
@Module
public abstract class ActivityBuilder {

    @ContributesAndroidInjector(modules = {
            TripActivityModule.class,
            TripDetailsProvider.class})
    abstract TripActivity bindTripActivity();

}

File: kotlinapp/src/main/java/com/tripplanner/di/ActivityBuilder.kt
package com.tripplanner.di

import com.tripplanner.ui.trip.TripActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Created by suyashg on 21/06/18.
 */
@Module
internal abstract class ActivityBuilder {
    @ContributesAndroidInjector(modules = [MainModule::class])
    internal abstract fun contributeMainInjector(): TripActivity
}